scripts/apply_style.py

```diff
@@ -346,6 +350,22 @@ def print_diff_and_fail():
         sys.exit(1)
 
 
+def find_names(root, names):
```

Comment:
I'd rather these changes were reverted, or at least tied more neatly into existing code. If the intent was to restrict the reformatting to a filesystem subtree, it's as simple as plumbing in the cwd into the `walk_source_directories` generator function instead of the `PROJECT_ROOT` constant. The rest of these changes are unnecessary. How many files with forbidden write access do we expect to have files with no read or write access in a local working copy? I'm also not happy with introducing an execution path where `commit` is hijacked to be a list of files instead of a string specifying a Git ref. Apart from the fact that this is not what `commit` is for, this kind of type ambiguities and reliance on `isinstance` makes the code hard to follow and maintain. Just because Python allows it doesn't mean it's a good idea.

Reply:
@wkaluza As I remember, these changes turned out necessary as otherwise, at some moment of time, apply_style.py wasn't able to find any file at all to process, unsure how it happened. Btw, I've added (slightly incomplete) path support in the command line but then it somehow disappeared from the script again few months ago.

Reply:
Seeing that no one else reported issues, this appears to be a problem local to your machine, and that's where I would seek to resolve it. In any case, these changes have nothing to do with this PR, so rather than clutter up this discussion, please revert.